open-api/lang-parser/src/main/java/io/deephaven/lang/parse/ParsedDocument.java

```diff
@@ -279,35 +282,35 @@ private Position findFromNodes(DocumentRange replaceRange, Node startNode, Node
     }
 
     private boolean startsBefore(Node kid, Position start) {
-        return kid.jjtGetFirstToken().positionStart().lessOrEqual(start);
+        return LspTools.lessOrEqual(kid.jjtGetFirstToken().positionStart(), start);
     }
 
     private Node refineEndNode(DocumentRange replaceRange, Node endNode) {
         if (endNode.jjtGetNumChildren() == 0) {
             return endNode;
         }
         for (Node kid : MappedIterable.reversed(endNode.getChildren())) {
-            if (startsBefore(kid, replaceRange.end)) {
+            if (startsBefore(kid, replaceRange.getEnd())) {
                 return refineEndNode(replaceRange, kid);
             }
         }
         return endNode;
     }
 
     @SuppressWarnings("Duplicates")
-    private Position findFromTokens(DocumentRange replaceRange, Token start, Token end) {
+    private Position.Builder findFromTokens(DocumentRange replaceRange, Token start, Token end) {
         // while it would be nice to actually iterate backwards here,
         // we only maintain forward links, and it would be a hassle / O(n) operation
         // to setup backlinks.
-        final Position startPos = start.positionStart();
-        final Position endPos = end.positionStart();
+        final Position.Builder startPos = start.positionStart();
+        final Position.Builder endPos = end.positionStart();
         // both asserts are >= because both start and end are the earliest token-containing-our-range we could find.
-        assert replaceRange.start.greaterOrEqual(startPos);
-        assert replaceRange.end.greaterOrEqual(endPos);
+        assert LspTools.greaterOrEqual(replaceRange.getStart(), startPos);
+        assert LspTools.greaterOrEqual(replaceRange.getEnd(), endPos);
```

Comment:
jvm code should use Assert or Require if this is a concern

Reply:
it's really not, just there for testing. Plus, it's possible for these types to someday live on a client, if that's ever something we'd want to bother w/.